apps/avatax/src/modules/avatax/avatax-client.ts

```diff
@@ -36,23 +39,52 @@ export class AvataxClient {
   constructor(private client: Avatax) {}
 
   async createTransaction({ model }: CreateTransactionArgs) {
-    return fromPromise(
-      /*
-       * We use createOrAdjustTransaction instead of createTransaction because
-       * we must guarantee a way of idempotent update of the transaction due to the
-       * migration requirements. The transaction can be created in the old flow, but committed in the new flow.
-       */
-      this.client.createOrAdjustTransaction({
-        model: { createTransactionModel: model },
-      }),
-      (error) => {
-        const parsedError = this.errorParser.parse(error);
-
-        /**
-         * TODO: Refactor errors so we are able to print error only for unhandled cases, otherwise use warnings etc
-         */
-        this.logger.error("Error calculating taxes", { error: parsedError });
-        return parsedError;
+    return appInternalTracer.startActiveSpan(
+      "calling AvaTax createOrAdjustTransaction API",
+      {
+        kind: SpanKind.CLIENT,
+        attributes: {
+          [ATTR_PEER_SERVICE]: "avatax",
+          "avatax.document_type": DocumentType[model.type ?? "Any"],
+        },
+      },
+      (span) => {
+        return fromPromise(
+          /*
+           * We use createOrAdjustTransaction instead of createTransaction because
+           * we must guarantee a way of idempotent update of the transaction due to the
+           * migration requirements. The transaction can be created in the old flow, but committed in the new flow.
+           */
+          this.client.createOrAdjustTransaction({
```

Comment:
isnt it already instrumented via http instrumentation?

Reply:
It is but we want to send those as client spans in the future